lib/two_factor_authentication/models/two_factor_authenticatable.rb

```diff
@@ -87,6 +103,19 @@ def generate_totp_secret
           ROTP::Base32.random_base32
         end
 
+        def create_backup_codes(options = {})
+          count = options[:count] || self.class.backup_code_count || 10
+          digits = options[:length] || self.class.backup_code_length || 8
+          backup_codes = (1..count).map do
+            random_base10(digits)
```

Comment:
- **AOL**: numeric only, 16 characters in groups of 4, separated by a dash
- **Dropbox**: alphanumeric, 16 characters in groups of 4, separated by a space
- **GitHub**: alphanumeric, 10 characters in groups of 5, separated by a dash
- **Heroku**: alphanumeric, 16 characters, all lowercase
- **Hover**: alphanumeric, 16 characters in groups of 4, separated by a space
- **IFTTT**: numeric only, 16 characters in groups of 4, separated by a space
- **Mailchimp**: alphanumeric, 16 characters, all caps
- **Stripe**: alpha only, 20 characters, 5 groups of 4, separated by dash